src/transformers/utils/tensor_parallel_utils.py

```diff
@@ -0,0 +1,147 @@
+from .. import BertPreTrainedModel, GPT2PreTrainedModel, TransfoXLPreTrainedModel
+
+
+# ie. nn.Linear(3 * dim, dim)
+# only defined the models that have fused attention layer.
+FUSED_ATTENTION_MAPPING = {
+    GPT2PreTrainedModel: {"attn.c_attn": 3, "crossattention.c_attn": 2},
+    TransfoXLPreTrainedModel: {"qkv_net": 3},
+}
+
+# ie. nn.Linear(out_dim, in_dim) or Conv1D()
+# only defined the models that have reversed parameters.
+REVERSED_PARAM_MAPPING = {
```

Comment:
I have already done tensor parallelization of 70 models in parallelformers, there are only two models that use this method in the transformers now.